update-version

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+
+set -e
+
+# Whitespace-separated list of components of the form 'service|image|tag'.
+COMPONENTS="\
+    flow-manager|gpii/universal|latest \
+    preferences-dataloader|gpii/preferences-dataloader|latest \
+    preferences|gpii/universal|latest \
+"
+
+for component in $COMPONENTS ; do
+    service=$(echo "$component" | awk -F'|' '{print $1}')
+    image=$(echo "$component" | awk -F'|' '{print $2}')
+    tag=$(echo "$component" | awk -F'|' '{print $3}')
+    echo "Looking up '$image:$tag' for service '$service'..."
+    api_token=$(\
+        curl -L -s \
+        "https://auth.docker.io/token?service=registry.docker.io&scope=repository:$image:pull" \
+        | jq -r .token \
+    )
+    # This idea is from
+    # https://stackoverflow.com/questions/39375421/can-i-get-an-image-digest-without-downloading-the-image.
+    api_out=$(\
+        curl -L -s \
+        --head \
+        -H "Authorization: Bearer $api_token" \
+        -H "Accept: application/vnd.docker.distribution.manifest.v2+json" \
+        https://index.docker.io/v2/$image/manifests/$tag \
+    )
+    # The 'tr' is because curl outputs headers using network line endings (\r\n)
+    # and our tools expect UNIX line endings (\n).
+    digest=$(\
+        echo "$api_out" \
+        | grep 'Docker-Content-Digest' \
+        | awk '{print $2}' \
+        | tr -d '\r' \
+    )
+    if [ -z "$digest" ] ; then
+        echo "Could not calculate digest for $image:$tag. Aborting."
+        echo "Here was the output from the API:"
+        echo "$api_out"
+        exit 1
+    else
+        echo "...found digest '$digest'."
+    fi
+
+    yaml="$yaml\n$service: \"$image@$digest\""
+done
+
+# Manual YAML construction is silly, but it avoids pulling in a dependency on a yaml library.
+outfile="version.yml"
+echo "Success! Writing $outfile..."
+echo "# This file is generated by https://github.com/gpii-ops/gpii-version-updater." > $outfile
+echo "---" >> $outfile
+# Note that '-e' doesn't work with OS X (BSD) echo, which emits a literal '-e'
+# instead :3. Without '-e', GNU echo prints literal '\n' :3 :3.
+echo -e "$yaml" | sort >> $outfile
```

Comment:
We don't have a policy on this but I usually run shell scripts through shellcheck:

```
Line 29:
        https://index.docker.io/v2/$image/manifests/$tag \
                                   ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                    ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 48:
    yaml="$yaml\n$service: \"$image@$digest\""
               ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
 
Line 58:
echo -e "$yaml" | sort >> $outfile
     ^-- SC2039: In POSIX sh, echo flags are undefined.
```